### PR TITLE
11.0 MIG web_domain_field

### PR DESCRIPTION
**Pull request by Maartincm**
Hello there.
This was an easy one. Pretty straightforward...

### CODE REVIEW

File: web_domain_field/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu/',
+    'depends': [
+        'web'
+    ],
+    'data': [
+        'views/web_domain_field.xml',
+    ],
+    'demo': [
```
Comment (tarteo) [inline web_domain_field/__manifest__.py]: Can you remove the empty key

**Review by tarteo — CHANGES_REQUESTED** [review]
Thanks. Code looks good to me. Is there anywhere I can test the functionality?

**Comment by Maartincm** [comment]
I guess I could write some testing routines as soon as I have any free time.
In the meantime, there is no other way to test this other than modifiyng some existent code locally, sorry.

**Comment by gabrielo77** [comment]
Hi @tarteo! What is needed to get this PR merged?

File: web_domain_field/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Web Domain Field',
+    'summary': """
+        Use computed field as domain""",
+    'version': '11.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu/',
```
Comment (nikul-serpentcs) [inline web_domain_field/__manifest__.py:11]: @Maartincm URL Should be [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Comment by nikul-serpentcs** [comment]
@Maartincm Use new README by fragment system, copying https://github.com/OCA/maintainer-tools/tree/master/template/module/readme and editing or removing the needed sections. You can then run locally oca-gen-addon-readme if you have OCA/maintainer-tools installed and you want to pregenerate the README.rst file (preferred), or keep previous README.rst file.

**Comment by Maartincm** [comment]
@nikul-serpentcs I ran the `oca-gen-addon-readme` script on the module and extracted the old **Usage** part into the `readme/USAGE.rst` file.
If there is something else that needs to be done for this PR to get merged, please let me know.

File: web_domain_field/static/src/js/pyeval.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+var original_pyeval = pyeval.eval;
+var original_ensure_evaluated = pyeval.ensure_evaluated;
+var py = window.py;
+
+/** copied from pyeval and not modified but required since not publicly
+exposed by web.pyeval**/
+
+// recursively wraps JS objects passed into the context to attributedicts
+// which jsonify back to JS objects
+function wrap(value) {
```
Comment (tarteo) [inline web_domain_field/static/src/js/pyeval.js:17]: Can make a PR in the Odoo repository to expose the functions?

**Review by tarteo — CHANGES_REQUESTED** [review]
Can you increase the coverage and fix the jslint issues. 

Change the module's website to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

Should be nice to have an example module like done with web_widget_x2many_2d_matrix (web_widget_x2many_2d_matrix_example) so it's easier to test the functionality.

File: web_domain_field/__manifest__.py (hunk 2 above)
Comment (NL66278) [reply web_domain_field/__manifest__.py:11]: @Maartincm please change the URL according to the request from @nikul-serpentcs

**Review by NL66278 — APPROVED** [review]
:+1: LGTM and tested. Not stopping approval, but would be good to apply the suggested change to the website key in the manifest file.

File: web_domain_field/static/src/js/pyeval.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        return eval_domains(object, context);
+    default:
+        return original_pyeval(type, object, context, options);
+     }
+}
+
+// override sync_eval in order to call our specialized implementation of
+// eval_domains
+function sync_eval_domains_and_contexts (source) {
+    var contexts = ([session.user_context] || []).concat(source.contexts);
```
Comment (Tardo) [inline web_domain_field/static/src/js/pyeval.js:219]: [session.user_context] never evaluates to false

**Review by Tardo — CHANGES_REQUESTED** [review]

File: web_domain_field/static/src/js/pyeval.js [hunk 5]
```diff
@@ -0,0 +1,232 @@
+odoo.define('web.domain_field', function (require) {
```
Comment (daramousk) [inline web_domain_field/static/src/js/pyeval.js:1]: Is there a reason this is not `web_domain_field.domain_field`?

**Review by daramousk — APPROVED** [review]
Hello and thank you for this, a lot of hacky stuff on the js side and I was hoping for a better way, but it seems stuff does not get exported so resorting to this makes sense.

This seems to be working great for me, I am not sure there will be need for this later on though, or this could be something that can be pushed as a PR to odoo.git itself.

**Comment by lmignon** [comment]
For the record, this change has been proposed to Odoo a long time ago but without success... IMO this use case should be supported natively. Moreover, the change is very small into odoo but we need to override a lot of code to implement this functionality outside odoo since parts of the required code is not exposed.. https://github.com/odoo/odoo/pull/17328/files

**Comment by NL66278** [comment]
@pedrobaeza Is it possible to remove the Needs Review tag?
@Maartincm and please change the website key in the manifest

Otherwise can one of the maintainers of this repo add a commit to change the webiste key?

Really need this one to be merged. Also for 12.0 by the way.

**Review by AaronHForgeFlow — APPROVED** [review]
Just functional tests

**Pull request closed without merging**